library/src/main/java/com/github/instacart/ahoy/Visit.java

```diff
@@ -69,7 +71,8 @@ public boolean isValid() {
     }
 
     public Visit withUpdatedExtraParams(Map<String, Object> extraParams) {
-        Map<String, Object> map = new HashMap<String, Object>(extraParams());
+        Map<String, Object> map = new ArrayMap<>();
+        map.putAll(extraParams());
```

Comment:
@mataanin I'm not sure I follow. Isn't the functionality the same as before? The `HashMap<>(Map<>)` constructor is a convenience function for creating a new map then adding the previous map's params.